<launch>
	<arg name="numero_maquinas"/>



	<include file="$(find tfg_javgarfer)/launch/image_procs.launch" if="$(eval arg('numero_maquinas') - 1 > 0)">
       <arg name="numero_maquinas" value="$(eval arg('numero_maquinas') - 1)"/>
   </include>



	<node machine="m$(arg numero_maquinas)" name="republish" pkg="image_transport" type="republish" ns="/stereo/$(arg numero_maquinas)/left" 
		args="compressed in:=image_raw raw out:=image_raw" ></node>


	<node  machine="m$(arg numero_maquinas)" name="republish" pkg="image_transport" type="republish" ns="/stereo/$(arg numero_maquinas)/right" 
		args="compressed in:=image_raw raw out:=image_raw" ></node>


	<node  machine="m$(arg numero_maquinas)" name="stereo_image_proc$(arg numero_maquinas)" pkg="stereo_image_proc" type="stereo_image_proc" ns="/stereo/$(arg numero_maquinas)">
		<param name="queue_size" value="1"/>
		<param name="prefilter_size" value="9"/>
		<param name="prefilter_cap" value="35"/>
		<param name="correlation_window_size" value="21"/>
		<param name="min_disparity" value="32"/>
		<param name="disparity_range" value="128"/>
		<param name="speckle_size" value="1000"/>
		<param name="approximate_sync" value="true" />


		<remap from="/stereo/$(arg numero_maquinas)/left/image_rect" to="/stereo/left/image_rect" />
		<remap from="/stereo/$(arg numero_maquinas)/right/image_rect" to="/stereo/right/image_rect" />

		<remap from="/stereo/$(arg numero_maquinas)/left/image_rect/compressed" to="/stereo/left/image_rect/compressed" />
		<remap from="/stereo/$(arg numero_maquinas)/right/image_rect/compressed" to="/stereo/right/image_rect/compressed" />

		<remap from="/stereo/$(arg numero_maquinas)/disparity" to="/stereo/disparity" />
 		<remap from="/stereo/$(arg numero_maquinas)/points2" to="/stereo/right/points2" />

	</node>


</launch>